presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -344,6 +345,59 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                 preparedStatements);
     }
 
+    public Session applyDefaultProperties(Map<String, String> systemPropertyDefaults, Map<String, Map<String, String>> catalogPropertyDefaults)
```

Comment:
applyDefaultProperties to me indicates that you are going to modify the instance. Perhaps something like `withDefaultProperties` is more indicative that it returns a new object.